matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -88,13 +92,103 @@ pub struct WebRtcSocketConfig {
     /// or: `wss://matchbox.example.com/your_game?next=2`
     ///
     /// The last form will pair player in the order they connect.
-    pub room_url: String,
+    pub(crate) room_url: String,
     /// Configuration for the (single) ICE server
-    pub ice_server: RtcIceServerConfig,
+    pub(crate) ice_server: RtcIceServerConfig,
     /// Configuration for one or multiple reliable or unreliable data channels
-    pub channels: Vec<ChannelConfig>,
+    pub(crate) channels: Vec<ChannelConfig>,
     /// The amount of attempts to initiate connection
-    pub attempts: Option<u16>,
+    pub(crate) attempts: Option<u16>,
+}
+
+impl WebRtcSocketBuilder {
+    /// Creates a new builder for a connection to a given room with the default ICE
+    /// server configuration, and three reconnection attempts.
+    ///
+    /// You must add at least one channel with [`WebRtcSocketBuilder::add_channel`],
+    /// [`WebRtcSocketBuilder::add_reliable_channel`], or
+    /// [`WebRtcSocketBuilder::add_unreliable_channel`] before you can build the
+    /// [`WebRtcSocket`]
+    pub fn new(room_url: impl ToString) -> Self {
+        Self {
+            room_url: room_url.to_string(),
+            ice_server: RtcIceServerConfig::default(),
+            channels: Vec::default(),
+            attempts: Some(3),
+        }
+    }
+
+    /// Sets the socket ICE server configuration.
+    pub fn ice_server(mut self, ice_server: RtcIceServerConfig) -> Self {
+        self.ice_server = ice_server;
+        self
+    }
+
+    /// Sets the number of attempts to make at reconnecting, if None the socket will
+    /// attempt to connect indefinitely.
```

Comment:
```suggestion
    /// Sets the number of attempts to make at reconnecting to the signalling server,
    /// if `None` the socket will attempt to connect indefinitely.
```
Should specify what we are reconnecting to (signaling server, not peers). Or make it obvious through naming.

Reply:
Yeah, that's a good call